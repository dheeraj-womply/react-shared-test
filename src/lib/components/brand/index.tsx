export { Logo } from './ui-elements'