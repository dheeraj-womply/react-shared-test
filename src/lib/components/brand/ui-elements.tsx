import React from 'react'
import styled from 'styled-components'
import { WOMPLY_LOGO, WOMPLY_LOGO_RED } from '../../resources/images'

export const LogoWrapper = styled.div`
    padding: 20px;
    img {
        width: 137px;
        height: 27px;
    }
`
interface LogoProps {
    alt?: string
    variant?: 'default' | 'red'
}

export const Logo: React.FC<LogoProps> = ({ alt, variant = 'default' }) => (
    <LogoWrapper>
        <img src={variant === 'red' ? WOMPLY_LOGO_RED : WOMPLY_LOGO} alt={alt}/>
    </LogoWrapper>
)
