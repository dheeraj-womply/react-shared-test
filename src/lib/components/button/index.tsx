export { Button } from './ui-elements'