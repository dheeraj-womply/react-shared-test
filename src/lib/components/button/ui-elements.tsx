import React from "react";
import {
    Button as MuiButton, ButtonProps,
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles({
    root: {
        fontSize: 14,
        fontWeight: 700,
        textTransform: 'uppercase',
    },
})

export const Button: React.FC<ButtonProps> = (props) => {
    const {
        color = 'primary',
        size = 'large',
        variant = 'contained',
        startIcon,
        ...rest
    } = props

    // eslint-disable-next-line react-hooks/rules-of-hooks
    const classes = props.classes ? { ...useStyles(), ...props.classes } : useStyles()

    return (
        <MuiButton
            {...rest}
            color={color}
            size={size}
            startIcon={startIcon}
            variant={variant}
            classes={classes}
        />
    )
}



