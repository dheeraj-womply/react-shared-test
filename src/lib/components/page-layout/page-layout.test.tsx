import React from 'react'
import renderer from 'react-test-renderer'
import PageLayout from './index'
import { Logo } from '../brand'
import { Button } from '../button'

test('Page Layout', () => {
  const component = renderer.create(
    <PageLayout
      header={
        <Logo alt="Womply Logo" variant="red" />
      }
      mainTitle="Oops. Something went wrong."
      mainHeader="Please reload the page to try again."
      mainSubheader="If this problem persists, submit a support ticket by visiting womply.com/contact-us."
      footer={
        <div style={{ padding: '12px 32px' }}>
          Brought to you by Womply,  a San Francisco-based technology company.
        </div>
      }
    >
      <div style={{ flex:1, display: 'flex', flexDirection: 'column', justifyContent: 'flex-end', paddingBottom: 40 }}>
        <Button size="large">
          Try Again
        </Button>
      </div>
    </PageLayout>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})