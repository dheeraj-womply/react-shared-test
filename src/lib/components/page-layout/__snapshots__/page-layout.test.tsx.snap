// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Layout 1`] = `
<div
  className="sc-AxjAm jThaSl"
>
  <header
    className="sc-AxirZ kEYvjm"
  >
    <div
      className="sc-AxgMl hieZL"
    >
      <img
        alt="Womply Logo"
        src="https://res.cloudinary.com/womply/image/upload/react-shared-components/womply-logo-red_tzbmeh_zqlcno.svg"
      />
    </div>
  </header>
  <main
    className="sc-AxiKw eiakxi"
  >
    <div
      className="sc-AxhUy fNfuCs"
    >
      Oops. Something went wrong.
    </div>
    <div
      className="sc-AxhUy fNfuCs"
    >
      Please reload the page to try again.
    </div>
    <div
      className="sc-AxhUy fNfuCs"
    >
      If this problem persists, submit a support ticket by visiting womply.com/contact-us.
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "paddingBottom": 40,
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-root-1 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Try Again
        </span>
      </button>
    </div>
  </main>
  <footer
    className="sc-AxhCb ixqYhm"
  >
    <div
      style={
        Object {
          "padding": "12px 32px",
        }
      }
    >
      Brought to you by Womply,  a San Francisco-based technology company.
    </div>
  </footer>
</div>
`;
